renderer/src/builder.rs

```diff
@@ -84,12 +84,13 @@ pub(crate) struct Occluder {
     pub(crate) coords: Vector2I,
 }
 
-impl<'a> SceneBuilder<'a> {
+impl<'a, L: RenderCommandListener> SceneBuilder<'a, L> {
     pub(crate) fn new(
         scene: &'a Scene,
         built_options: &'a PreparedBuildOptions,
-        listener: Box<dyn RenderCommandListener>,
-    ) -> SceneBuilder<'a> {
+        listener: L,
+    ) -> SceneBuilder<'a, L> {
+        let effective_view_box = scene.effective_view_box(built_options);
```

Comment:
I replaced the trait object with a generic, as in WebAssembly it is all single threaded and are no runtime decisions possible. I did not measure the performance gain.

Reply:
I used a trait object to improve compilation time and because the overhead of vtable dispatch should be minimal.